### DISCUSSION TITLE
How to specify a subtype for an extended class? (Not Abstract)

### DISCUSSION THREAD

**Discussion by demlik** [instancio/instancio#527] [Q&A]
Hi,

Given the following:

class Animal{}
class Dog extends Animal{}

Can I get Instancio to create a Dog when it sees an Animal? 

Thanks

**Comment by armandino** [comment]
@demlik yes, should be possible using subtype mapping. Please take a look at the  [user guide](https://www.instancio.org/user-guide/#subtype-mapping) (there are a couple of examples).
